src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -1036,14 +978,14 @@ public ValueGetter<T> GetNAOrDefaultGetter<T>(ColumnType type)
         public void Convert(ref BL src, ref SB dst)
```

Comment:
I'm not sure if this is the source of the text-saver bug or a separate bug, but either way this ought to be fixed as well.

Reply:
Do you mean the two functions above? 

---
In reply to: [216129079](https://github.com/dotnet/machinelearning/pull/863#discussion_r216129079) [](ancestors = 216129079)

Reply:
Probably. I am trying to give you a hint of where your bug is, however I need you to fix it.

---
In reply to: [216184946](https://github.com/dotnet/machinelearning/pull/863#discussion_r216184946) [](ancestors = 216184946,216129079)

Reply:
I fixed the bug even before I read this comment but got confused after reading it so I asked. I believe the bug is in the functions above this one, where we need to print special value for NaN so that when they are converted back to float/double they are parsed as NaN instead of default.

Thanks for finding the float/double to text bug. 

---
In reply to: [216192538](https://github.com/dotnet/machinelearning/pull/863#discussion_r216192538) [](ancestors = 216192538,216184946,216129079)